supervisor/addons/model.py

```diff
@@ -538,14 +539,13 @@ def need_build(self) -> bool:
         return ATTR_IMAGE not in self.data
 
     @property
-    def map_volumes(self) -> dict[str, bool]:
-        """Return a dict of {volume: read-only} from add-on."""
+    def map_volumes(self) -> dict[MappingType, FolderMapping]:
+        """Return a dict of {MappingType: FolderMapping} from add-on."""
         volumes = {}
         for volume in self.data[ATTR_MAP]:
-            result = RE_VOLUME.match(volume)
-            if not result:
-                continue
-            volumes[result.group(1)] = result.group(2) != "rw"
+            volumes[MappingType(volume.get("type"))] = FolderMapping(
+                volume.get("path"), volume.get("read_only")
+            )
```

Comment:
```suggestion
            volumes[MappingType(volume[ATTR_TYPE)] = FolderMapping(
                volume.get(ATTR_PATH), volume[ATTR_READ_ONLY]
            )
```
Since type is required, should not use `get`. Also `MappingType(None)` isn't valid and would fail anyway so using `get` doesn't stop an error here. And the schema should default `read_only` to `True` as suggested in my other comment so that ensures it always exists as well. Plus if `read_only` was actually set to `None` it would break here:
https://github.com/home-assistant/supervisor/blob/40bf7c081b4ad38fdfe661ce0ca88da2a770e09b/supervisor/docker/addon.py#L348